cpp/src/arrow/builder.cc

```diff
@@ -736,7 +736,7 @@ Status BooleanBuilder::AppendValues(const std::vector<bool>& values,
 
   int64_t i = 0;
   internal::GenerateBitsUnrolled(raw_data_, length_, length,
-                                 [values, &i]() -> bool { return values[i++]; });
+                                 [&values, &i]() -> bool { return values[i++]; });
```

Comment:
Why?

Reply:
I don't know. This is crashing in the release version when it tries to copy the "values" vector into closure. It's also a good change, we don't want to make a copy in the first place.

Reply:
Ah, nice catch indeed.